### DISCUSSION TITLE
Cannot find module 'src/web/utils/environmentVariablesSetup' Require stack:

### DISCUSSION THREAD

**Discussion by JagadeeshC78** [webdriverio/webdriverio#13801] [Community]
I'm getting the following error while executing specs in VS Code. This is as part of #WebdriverIO automation.

Everything was working file till i updated node/npm/wdio. All the specs and files are in place. Can share more details if necessary. Thanks in advance for your help.

% npx wdio run src/config/wdio.conf.ts --spec ./src/web/tests/pre-req/pre-requisite.spec.ts 2024-10-25T03:01:44.883Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts: Cannot find module 'src/web/utils/environmentVariablesSetup' Require stack:

/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts Error: Cannot find module 'src/web/utils/environmentVariablesSetup' Require stack:
/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1144:15) at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/jagadeeshchandrasekaran/wdio/webdriverio/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30) at Function.Module._load (node:internal/modules/cjs/loader:985:27) at Module.require (node:internal/modules/cjs/loader:1235:19) at require (node:internal/modules/helpers:176:18) at Object. (/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts:7:1) at Module._compile (node:internal/modules/cjs/loader:1376:14) at Module.m._compile (/Users/jagadeeshchandrasekaran/wdio/webdriverio/node_modules/ts-node/src/index.ts:1618:23) at Module._extensions..js (node:internal/modules/cjs/loader:1435:10) at Object.require.extensions. [as .ts] (/Users/jagadeeshchandrasekaran/wdio/webdriverio/node_modules/ts-node/src/index.ts:1621:12) { code: 'MODULE_NOT_FOUND',
% npm run test-PREMACGCSYSAD

webdriveriomochaframework@1.0.0 test-PREMACGCSYSAD npm run pretest && npm run remove-allure-results-folder && ENV=PreRequisite_SysAd_Mac_Eng_GC wdio src/config/wdio.conf.ts

webdriveriomochaframework@1.0.0 pretest node killPort.js webdriveriomochaframework@1.0.0 remove-allure-results-folder node delAllure.js

Directory deleted Selected Capabilities: [ { browserName: 'chrome', services: [ [Array] ], acceptInsecureCerts: true, 'goog:chromeOptions': { args: [Array], excludeSwitches: [Array], useAutomationExtension: false, prefs: [Object] } } ] Browser : Chrome Feature Type : PreReq ENV variable: PreRequisite_SysAd_Mac_Eng_GC Specs to run: [ './src/web/tests/pre-req/pre-requisite.spec.ts' ]

Execution of 0 workers started at 2024-10-25T02:42:29.930Z

2024-10-25T02:42:30.061Z ERROR @wdio/cli:launcher: No specs found to run, exiting with failure

**Comment by christian-bromann** [comment]
It seems like you are importing in your `wdio.conf.ts` (which is located at `file:///Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts`) a file with the path `src/web/utils/environmentVariablesSetup` which doesn't exist. You may want to try `../web/utils/environmentVariablesSetup`

**Reply by JagadeeshC78** [reply]
I've updated the same as suggested. Still seeing the same issue
jagadeeshchandrasekaran@JagadeeshC webdriverio % npx wdio run src/config/wdio.conf.ts --spec ./src/web/tests/pre-req/pre-requisite.spec.ts
2024-10-25T05:08:54.968Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts: Cannot read properties of undefined (reading 'browser')
TypeError: Cannot read properties of undefined (reading 'browser')
    at getBrowser (/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/web/utils/environmentVariablesSetup.ts:18:34)
    at Object.<anonymous> (/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/config/wdio.conf.ts:10:30)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module.m._compile (/Users/jagadeeshchandrasekaran/wdio/webdriverio/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/jagadeeshchandrasekaran/wdio/webdriverio/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1023:12)
    at cjsLoader (node:internal/modules/esm/translators:356:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:305:7)

jagadeeshchandrasekaran@JagadeeshC webdriverio % npm run test-PREMACGCSYSAD                                                               

> webdriveriomochaframework@1.0.0 test-PREMACGCSYSAD
> npm run pretest && npm run remove-allure-results-folder && ENV=PreRequisite_SysAd_Mac_Eng_GC wdio src/config/wdio.conf.ts


> webdriveriomochaframework@1.0.0 pretest
> node killPort.js


> webdriveriomochaframework@1.0.0 remove-allure-results-folder
> node delAllure.js

Directory deleted
Browser : Chrome
Feature Type : PreReq
ENV variable: PreRequisite_SysAd_Mac_Eng_GC
Specs to run: [ './src/web/tests/pre-req/pre-requisite.spec.ts' ]

Execution of 0 workers started at 2024-10-25T05:09:08.286Z

**Reply by JagadeeshC78** [reply]
var allure = require('allure-commandline');
declare var require: any
const { execSync, exec } = require('child_process');

import loginPage from '../web/pages/login/login.page';
import { multiCapabilities } from "./capabilities";
import { setSiteEnvironment, getBrowser, getObjective, getLanguage, getUserCredentials } from '../web/utils/environmentVariablesSetup';

**Reply by JagadeeshC78** [reply]
@christian-bromann appreciate your help to resolve this issue asap. Anything further i could check??

**Reply by christian-bromann** [reply]
Sorry I can't help with such specific things, please read the error message and debug further:

> TypeError: Cannot read properties of undefined (reading 'browser')
at getBrowser (/Users/jagadeeshchandrasekaran/wdio/webdriverio/src/web/utils/environmentVariablesSetup.ts:18:34)
